lib/rules/destructuring-assignment.js

```diff
@@ -125,13 +136,29 @@ module.exports = {
         const SFCComponent = components.get(context.getScope(node).block);
 
         const destructuring = (node.init && node.id && node.id.type === 'ObjectPattern');
+        const identifier = (node.init && node.id && node.id.type === 'Identifier');
         // let {foo} = props;
-        const destructuringSFC = destructuring && (node.init.name === 'props' || node.init.name === 'context');
+        const destructuringSFC = destructuring && node.init.name === 'props';
+        // let {foo} = useContext(aContext);
+        const destructuringUseContext = destructuring && node.init.callee && node.init.callee.name === 'useContext';
+        // let foo = useContext(aContext);
+        const assignUseContext = identifier && node.init.callee && node.init.callee.name === 'useContext';
```

Comment:
are we sure `useContext` is the react hook? I think we should make sure that nobody's using a custom hook or function named `useContext`, and that this `useContext` is imported/required from `react`.

Reply:
https://reactjs.org/docs/hooks-reference.html#usecontext
here in official docs of react says ``useContext`` is one of basic hooks with ``useState``, ``useEffect``.
and we import ``useContext`` like ``import {useContext} from 'react'``;

I think if somebody's using function named ``useContext`` which overwrite basic hooks , I think we should make another rule about custom hooks which overwrite basic hooks.

Reply:
Someone might be using a version of React that doesn't have hooks.

Reply:
then is there any possible way to check version of react? or we should not include this part.

Reply:
Yes! Check out `lib/util/version.js`'s `testReactVersion` method.

Reply:
do I have to import ``testReactVersion`` method in ``lib/rules/destructuring-assignment'`` to check react version in rules? I have no idea.. could you help me with this?

Reply:
Yes, that's right